### PR TITLE
Upgrade web scrapers

### PR DESCRIPTION
**Pull request by BenTalese**

### CODE REVIEW

File: framework/merchant_api/infrastructure/merchant_data_providers/coles_provider.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            _BuildID = json.loads(_Soup.find(id='__NEXT_DATA__').contents[0])['buildId']
+
+            _Url = f'https://www.coles.com.au/_next/data/{_BuildID}/en/search.json'
+            _Params = {
+                'q': search_term,
+                'page': 1
+            }
+
+            _ScrapedProductOffers = []
+            while True:
```
Comment (BenTalese) [inline framework/merchant_api/infrastructure/merchant_data_providers/coles_provider.py:87]: for all providers, may want to add safety net for infinite spam, maybe a time based back off?

File: framework/web_app/src/pages/ProductSearch.vue [hunk 2]
```diff
@@ -129,14 +129,14 @@ const productStore = useProductStore();
 const searchTerm = ref('');
 const previousSearchTerm = ref<undefined | string>();
 
-let currentPage = 1;
+let currentPage = 1; // TODO: What to do with this now?
```
Comment (BenTalese) [inline framework/web_app/src/pages/ProductSearch.vue:132]: thinking just delete

File: framework/merchant_api/infrastructure/service_collection_builder.py [hunk 3]
```diff
@@ -16,15 +22,36 @@ def __init__(self, service_provider: DependencyInjectorServiceProvider):
...
+
+        logger = logging.getLogger(__name__)
+        log_filename = log_folder / 'mapi_logs.txt'
+        file_handler = TimedRotatingFileHandler(log_filename, when="midnight", interval=1, backupCount=30)
+        file_handler.setFormatter(logging.Formatter('%(asctime)s | %(levelname)s | %(lineno)04d | %(message)s'))
+        logger.setLevel(_ConfigurationManager.get_log_level())
+        logger.addHandler(file_handler)
+
+        # FIXME: Clapy needs update to allow overriding the name of the service
+        # self.service_provider.register_service(providers.Object, logger)
```
Comment (BenTalese) [inline framework/merchant_api/infrastructure/service_collection_builder.py:50]: gotta deal with this

File: framework/web_app/src/stores/productStore.ts [hunk 4]
```diff
@@ -42,7 +42,9 @@ export const useProductStore = defineStore('product', () => {
      * @returns The new length of the stores array after adding the new merchants.
      */
     const addStoresToProductSearchFilter = (merchantsData: Merchant[]): number =>
-        productSearchOfferFilters.stores.push(...merchantsData);
+        productSearchOfferFilters.stores.push(
+            ...merchantsData.filter(merchant => merchant.is_enabled)
```
Comment (atiftonat) [inline framework/web_app/src/stores/productStore.ts:46]: This is just a general 'add' method.

IMO either:
- Filter the merchants argument to be filter by is enabled in the merhcantStore where this method is invoked
- OR re-purpose this method to not be a general add and rename it. E.g., addEnabledStoresToProductSearchFilter

File: framework/web_app/src/pages/ProductSearch.vue [hunk 5]
```diff
@@ -11,7 +11,7 @@
 
     <div v-show="showFiltersContainer" class="no-wrap row q-pa-sm scroll scrollbar-none">
 
-        <select-component label="Stores" :multiple="true" :options="merchantStore.merchants"
+        <select-component label="Stores" :multiple="true" :options="merchantStore.merchants.filter(m => m.is_enabled)"
```
Comment (atiftonat) [inline framework/web_app/src/pages/ProductSearch.vue:14]: Although a small dataset this filter is going to apply every time this page or component is rendered.

Would recommend creating a computed method in this file that is getEnabledMerchants or getEnabledStores.

**Review by atiftonat — COMMENTED** [review]
Minor comments. 
Skimmed over the backend but looks amazing, great work!

File: framework/web_app/src/pages/ProductSearch.vue (hunk 5 above)
Comment (BenTalese) [reply framework/web_app/src/pages/ProductSearch.vue]: fixed :D using method from store now

File: framework/web_app/src/stores/productStore.ts (hunk 4 above)
Comment (BenTalese) [reply framework/web_app/src/stores/productStore.ts]: fixed :D using common method from store and moved this logic to there so this method is now dumb

**Pull request merged**